internal/services/securitycenter/security_center_server_vulnerability_assessment_virtual_machine_resource.go

```diff
@@ -0,0 +1,184 @@
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.VirtualMachineID,
+			},
+		},
+	}
+}
+
+func resourceServerVulnerabilityAssessmentVirtualMachineCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	virtualMachineId, err := computeParse.VirtualMachineID(d.Get("virtual_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	vulnerabilityAssessment, err := client.Get(ctx, virtualMachineId.ResourceGroup, computeProvider, vmType, virtualMachineId.Name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(vulnerabilityAssessment.Response) {
+			return fmt.Errorf("checking for presence of existing Advanced Threat Protection for %s: %+v", *virtualMachineId, err)
+		}
+	}
+
+	if vulnerabilityAssessment.ID != nil && *vulnerabilityAssessment.ID != "" {
+		return tf.ImportAsExistsError("azurerm_security_center_server_vulnerability_assessment_virtual_machine", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, virtualMachineId.ResourceGroup, computeProvider, vmType, virtualMachineId.Name)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %s: %+v", *virtualMachineId, err)
+	}
+
+	id := parse.NewVulnerabilityAssessmentVmID(virtualMachineId.SubscriptionId, virtualMachineId.ResourceGroup, virtualMachineId.Name, "Default")
+	d.SetId(id.ID())
+
+	timeout, _ := ctx.Deadline()
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:      []string{"Pending"},
+		Target:       []string{"Succeeded"},
+		Refresh:      serverVulnerabilityAssessmentVirtualMachineStateRefreshFunc(ctx, client, id.ResourceGroup, id.VirtualMachineName),
+		PollInterval: 10 * time.Second,
+		Timeout:      time.Until(timeout),
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for the completion of the creating/updating of %s: %+v", id, err)
+	}
+
+	return resourceServerVulnerabilityAssessmentVirtualMachineRead(d, meta)
+}
+
+func resourceServerVulnerabilityAssessmentVirtualMachineRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.VulnerabilityAssessmentVmID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	var virtualMachineId *computeParse.VirtualMachineId
+	if id.ID() != "" {
+		newVirtualMachineId := computeParse.NewVirtualMachineID(id.SubscriptionId, id.ResourceGroup, id.VirtualMachineName)
+		virtualMachineId = &newVirtualMachineId
+	} else {
+		newVirtualMachineId, err := computeParse.VirtualMachineID(d.Get("virtual_machine_id").(string))
+		if err != nil {
+			return err
+		}
+		virtualMachineId = newVirtualMachineId
+	}
```

Comment:
(as above) we can assume this parses correctly?

```suggestion
	virtualMachineId := computeParse.NewVirtualMachineID(id.SubscriptionId, id.ResourceGroup, id.VirtualMachineName)
```